### PR TITLE
Merge SSO changes into dev

### PR DESCRIPTION
**Pull request by NatSquared**
Issue #: [🎟️DESENG-408](https://apps.itsm.gov.bc.ca/jira/browse/DESENG-408)

*Description of changes:*
Most of the configuration changes required to make MET function correctly when using the BC Government's hosted Single Sign On Keycloak servers.

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of the met-public license (Apache 2.0).

### CODE REVIEW

File: analytics-api/src/analytics_api/config.py [hunk 1]
```diff
@@ -28,20 +28,26 @@
...
-    return config
+    config_mapping = {
+        'development': DevConfig,
+        'default':   ProdConfig,
+        'staging':    ProdConfig,
+        'production': ProdConfig,
+        'testing':    TestConfig,
+    }
+    try:
+        print(f'Loading configuration: {environment}...')
```
Comment (ShawnTurple) [inline analytics-api/src/analytics_api/config.py:47]: is print the logging mechanism for this application, no logging plugin or code

Comment (NatSquared) [reply analytics-api/src/analytics_api/config.py:47]: Print was the logging mechanism previously used in this file. I can upgrade to `current_app.logger.debug` :)

**Review by ShawnTurple — APPROVED** [review]
This is just reviewing previous commits to be merged into main branch.
Just a final review or look of code, no in depth look.

File: analytics-api/src/analytics_api/config.py (hunk 1 above)
Comment (NatSquared) [reply analytics-api/src/analytics_api/config.py:47]: Nevermind; I cannot. `current_app` is only defined after the configuration is set up, and that is done in this file. I can print to STDERR

File: met-api/src/met_api/services/engagement_service.py [hunk 2]
```diff
@@ -99,7 +97,7 @@ def _get_scope_options(user_roles, has_team_access):
             return EngagementScopeOptions(restricted=False)
         if has_team_access:
             # return those engagements where user has access for edit members..
-            # either he has edit_member role or check if he is a team member
+            # either they have edit_member role or check if they are a team member
```
Comment (jareth-whitney) [inline met-api/src/met_api/services/engagement_service.py:100]: well done

**Review by jareth-whitney — APPROVED** [review]
Some great changes in here. I appreciate how everything is properly labelled and organized. Gender-neutral comments are always an improvement as well. Maybe resolve Shawn's point about print if applicable. I won't block you, approved :)

**Pull request merged**